### ISSUE TITLE
Build error when disabling twitch but enabling network

### ISSUE THREAD

**Issue by jansegre** [OpenRCT2/OpenRCT2#5469]
**Version:** all since 0.0.3 (probably)

**Steps to reproduce:**
1. when configuring cmake set `DISABLE_NETWORK=OFF` and `DISABLE_HTTP_TWITCH=ON`
2. compile

This seems to happen because of [this](https://github.com/OpenRCT2/OpenRCT2/blob/develop/CMakeLists.txt#L140), [this](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/network/http.h#L20) and [this](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/network/NetworkServerAdvertiser.cpp#L103). 

Basically it boils down to `DISABLE_HTTP_TWITCH=ON` setting the definition `DISABLE_HTTP` which is required for the network functionality that is still enabled. I didn't take the time to test if setting `DISABLE_HTTP` only if both twitch and network are disabled would fix the problem.

**Comment by janisozaur** [comment]
Good catch, although I don't think if it will be fixed properly.

These options exist primarily to aid in porting efforts and ideally, all platforms should target the "full" builds.

I'd rather add a check in CMake to verify that if curl ("HTTP") is disabled, openssl ("network") is also disabled, with appropriate warning. Someone else is welcome to provide a better fix, but keep in mind the incoming https://github.com/OpenRCT2/OpenRCT2/pull/5458

**Comment by rwjuk** [comment]
Personally, I'd say it'd be wise to postpone any fix for this (if deemed appropriate) to after #5458 is merged at least.

**Comment by janisozaur** [comment]
Fixed in https://github.com/OpenRCT2/OpenRCT2/pull/6760. It makes porting a bit easier.
